### DISCUSSION TITLE
Slow API query

### DISCUSSION THREAD

**Discussion by rbkayz** [supabase/supabase#9988] [Questions] [answered]
Hello community,

Facing an issue with an RPC function. 

I've checked the function performance thoroughly using explain analyze. It takes <70ms for the function to execute. However, the API response takes more than 5 sec. And there isn't a lot of data being transferred - I removed all the data in the response and it still takes over 5 sec

Any idea why this might be happening? Or a solve for it?

Thanks

**Comment by GaryAustin1** [comment, marked as answer]
I also commented on your Discord issue...
When your ran your explain analyze was it in the dashboard sql editor?  If so could be something different with RLS as that would be enforced thru the API (unless you are using service_role).   Did it perform better at one time?

**Reply by rbkayz** [reply]
You, ser, are a lifesaver. RLS is indeed the problem. Debugging the cause of the slow function query on RLS now

**Reply by GaryAustin1** [reply]
See this thread on what some of us are looking at regarding RLS....
https://github.com/supabase/supabase/discussions/9311
